shiksha-api/app-service/app/main.py

```
@@ -10,6 +10,17 @@
 from app.routers import chat_router, chat_router_mcp, presentation_router, question_paper_router
 
+# Initialize Langfuse instrumentation at import time.
+from app.observability.langfuse_setup import init_langfuse
+_langfuse_client = init_langfuse(app_env=settings.app_env)
+
+# Wire LlamaIndex → Langfuse so RAG-path LLM + embedding calls appear as child spans.
+try:
+    from llama_index.instrumentation.langfuse import LlamaIndexInstrumentor
+    LlamaIndexInstrumentor().instrument()
+except Exception:
+    pass  # no-op if langfuse keys absent or package missing
```

Ruff (0.15.18)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)